### ISSUE TITLE
Fix security context and container privilege issues

### ISSUE THREAD

**Issue by steven-zou** [goharbor/harbor-operator#127]
When deploying Harbor to the strict cluster, it will be failed.

run as nonroot
numeric nonroot user (<1000)

This fix is depending on the related fixes @Harbor side

**Comment by ywk253100** [comment]
Please refer to https://github.com/goharbor/harbor-helm/pull/782 to do the corresponding changes in the operator.

**Comment by ywk253100** [comment]
- Specify the security context:
```
  securityContext:
        runAsUser: 999
        fsGroup: 999
```

- Put the data of postgres in a sub folder

**Comment by heww** [comment]
Closed by #420 and #434
